src/main/java/org/apache/maven/plugins/scmpublish/AbstractScmPublishMojo.java

```diff
@@ -179,19 +179,22 @@ public abstract class AbstractScmPublishMojo extends AbstractMojo {
     protected ScmRepositoryConfigurator scmRepositoryConfigurator;
 
     /**
-     * The serverId specified in the settings.xml, which should be used for the authentication.
+     * The server id specified in the {@code settings.xml}, which should be used for the authentication.
+     * @see <a href="https://maven.apache.org/settings.html#servers">Settings Reference</a>
      */
-    @Parameter
+    @Parameter(property = "scmpublish.serverId", defaultValue = "${project.distributionManagement.site.id}")
```

Comment:
How is the behavior if you want this null/empty and supply username/password? We don't have a way to null prepoulated params.